crates/noirc_evaluator/src/ssa/context.rs

```diff
@@ -424,6 +471,190 @@ impl<'a> SsaContext<'a> {
         self[target_block].instructions.insert(1, phi_id);
         phi_id
     }
+
+    fn memcpy(&mut self, l_type: ObjectType, r_type: ObjectType) {
+        if l_type == r_type {
+            return;
+        }
+
+        if let (ObjectType::Pointer(a), ObjectType::Pointer(b)) = (l_type, r_type) {
+            let len = self.mem.arrays[a as usize].len;
+            let adr_a = self.mem.arrays[a as usize].adr;
+            let adr_b = self.mem.arrays[b as usize].adr;
+            for i in 0..len {
+                let idx_b = self.get_or_create_const(
+                    FieldElement::from((adr_b + i) as i128),
+                    ObjectType::Unsigned(32),
+                );
+                let idx_a = self.get_or_create_const(
+                    FieldElement::from((adr_a + i) as i128),
+                    ObjectType::Unsigned(32),
+                );
+                let load = self.new_instruction(idx_b, idx_b, Operation::Load(b), r_type);
+                self.new_instruction(load, idx_a, Operation::Store(a), l_type);
+            }
+        } else {
+            dbg!(&l_type);
+            dbg!(&r_type);
+            unreachable!("invalid type, expected arrays");
+        }
+    }
+
+    pub fn handle_assign(&mut self, lhs: NodeId, index: Option<NodeId>, rhs: NodeId) -> NodeId {
+        let lhs_type = self.get_object_type(lhs);
+        let rhs_type = self.get_object_type(rhs);
+        let i = 0;
+        let i_obj =
+            self.get_or_create_const(FieldElement::from(i as i128), ObjectType::Unsigned(32));
```

Comment:
It is not really useful because we have only one return value but it will be useful when we return struct/tuples to indicate which element we refer to in the return instruction.